librz/arch/isa/pyc/opcode_311.c

```diff
@@ -0,0 +1,114 @@
+	rm_op(.op_obj = ret->opcodes, .op_name = "BINARY_XOR", .op_code = 65);
+	rm_op(.op_obj = ret->opcodes, .op_name = "BINARY_OR", .op_code = 66);
+	// // Remove INPLACE_*
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_MATRIX_MULTIPLY", .op_code = 17);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_FLOOR_DIVIDE", .op_code = 28);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_TRUE_DIVIDE", .op_code = 29);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_ADD", .op_code = 55);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_SUBTRACT", .op_code = 56);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_MULTIPLY", .op_code = 57);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_MODULO", .op_code = 59);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_POWER", .op_code = 67);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_LSHIFT", .op_code = 75);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_RSHIFT", .op_code = 76);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_AND", .op_code = 77);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_XOR", .op_code = 78);
+	rm_op(.op_obj = ret->opcodes, .op_name = "INPLACE_OR", .op_code = 79);
+
+	rm_op(.op_obj = ret->opcodes, .op_name = "CALL_FUNCTION", .op_code = 131);
+	rm_op(.op_obj = ret->opcodes, .op_name = "CALL_FUNCTION_KW", .op_code = 141);
+	rm_op(.op_obj = ret->opcodes, .op_name = "CALL_METHOD", .op_code = 161);
+
+	rm_op(.op_obj = ret->opcodes, .op_name = "DUP_TOP_TWO", .op_code = 5);
+	rm_op(.op_obj = ret->opcodes, .op_name = "DUP_TOP", .op_code = 4);
+	rm_op(.op_obj = ret->opcodes, .op_name = "ROT_TWO", .op_code = 2);
+	rm_op(.op_obj = ret->opcodes, .op_name = "ROT_THREE", .op_code = 3);
+	rm_op(.op_obj = ret->opcodes, .op_name = "ROT_FOUR", .op_code = 6);
+
+	rm_op(.op_obj = ret->opcodes, .op_name = "JUMP_ABSOLUTE", .op_code = 113);
+	rm_op(.op_obj = ret->opcodes, .op_name = "POP_JUMP_IF_FALSE", .op_code = 114);
+	rm_op(.op_obj = ret->opcodes, .op_name = "POP_JUMP_IF_TRUE", .op_code = 115);
+
+	rm_op(.op_obj = ret->opcodes, .op_name = "SETUP_WITH", .op_code = 143);
+	rm_op(.op_obj = ret->opcodes, .op_name = "SETUP_ASYNC_WITH", .op_code = 154);
+
+	// // New Opcodes
+	def_op(.op_obj = ret->opcodes, .op_name = "ASYNC_GEN_WRAP", .op_code = 87, .pop = 1, .push = 1);
+	def_op(.op_obj = ret->opcodes, .op_name = "RETURN_GENERATOR", .op_code = 75, .pop = 0, .push = 0);
+	def_op(.op_obj = ret->opcodes, .op_name = "SEND", .op_code = 123, .pop = 1, .push = 1);
+	def_op(.op_obj = ret->opcodes, .op_name = "COPY_FREE_VARS", .op_code = 149, .pop = 0, .push = 0);
+	def_op(.op_obj = ret->opcodes, .op_name = "MAKE_CELL", .op_code = 135, .pop = 0, .push = 0);
+
+	def_op(.op_obj = ret->opcodes, .op_name = "CHECK_EXC_MATCH", .op_code = 36, .pop = 1, .push = 1);
+	def_op(.op_obj = ret->opcodes, .op_name = "CHECK_EG_MATCH", .op_code = 37, .pop = 9, .push = 1); // Conditional pushing and popping, I dont know how that is implemented in analysis.
```

Comment:
```suggestion
	def_op(.op_obj = ret->opcodes, .op_name = "CHECK_EG_MATCH", .op_code = 37, .pop = 9, .push = 1);
```

Seems to be resolved in opcode_analysis.c?

Reply:
I dont think so, I tried reading for some opcodes of previous versions that pushed and popped conditionally.
but didnt find something of use, and in opcode_*.c most of the time they are defined with pop=9 and push=1

Reply:
Conditional instructions are generally marked with the `op_type`. There is a `RZ_ANALYSIS_OP_TYPE_COND` bit to set. Even if it doesn't have much impact, you can set it. Just for the future. So we don't miss known information.

Reply:
oh ok